### PR TITLE
BUG: Fix postgres boolean sum and add summary functionality

### LINKED ISSUE #1035: Summary is broken on postgres due to issues summing booleans
Hi: Ibis is awesome, and I realize the postgres support is highly experimental - but I ran into a limitation that makes a lot of other tasks a lot more cumbersome.

Boolean types on postgres cannot be summed, so quite a bit of functionality is missing and require verbose work around.

The simplest example:

`a.clothing_size.isnull().sum().execute()`

Gives this output:

```
ProgrammingError: (psycopg2.ProgrammingError) function sum(boolean) does not exist
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
 [SQL: 'SELECT sum(t0.clothing_size IS NULL) AS sum \nFROM monster.asin_dimension AS t0']

```

Is this something that IBIS should handle under the hood, or users should write back-end aware code?

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #1035

### CODE REVIEW

**Comment by cpcloud** [comment]
I'm not sure this is necessarily the right approach. However, it's extensible and allows us to add arbitrary promotion rules for specific backends easily.

File: ibis/sql/postgres/compiler.py [hunk 1]
```diff
@@ -622,3 +637,26 @@ class PostgreSQLExprTranslator(alch.AlchemyExprTranslator):
...
+    PostgreSQLExprTranslator,
+    (ops.StandardDev, ops.Variance),
+    ir.BooleanColumn,
+    (ir.BooleanColumn, type(None)),
+    six.string_types
+)
+def upcast_variance_stddev_boolean_reduction(
+    _, operation, column, where, kind
+):
+    return type(operation)(column.cast('int32'), where, kind).to_expr()
```
Comment (wesm) [inline ibis/sql/postgres/compiler.py]: I don't have a strong view, but you like this more than putting the casting logic into the translator implementations for these operators? I guess if you start accumulating a lot of upcasting rules then this is a little easier

Comment (cpcloud) [inline ibis/sql/postgres/compiler.py]: Agree. I think I'll move this into the translation rule set for postgres. If we start to get a bunch of these we can always refactor to use this approach.

**Comment by cpcloud** [comment]
Merging on green.

**Pull request closed without merging**